src/interface/meshblock_data.cpp

```diff
@@ -53,10 +53,8 @@ void MeshBlockData<T>::Add(const std::string &label, const Metadata &metadata) {
     }
     int varIndex = metadata.GetSparseId();
     sparseMap_[label]->Add(varIndex, arrDims);
-    if (metadata.IsSet(Metadata::FillGhost)) {
-      auto &v = sparseMap_[label]->Get(varIndex);
-      v->allocateComms(pmy_block);
-    }
+    auto &v = sparseMap_[label]->Get(varIndex);
+    v->allocateComms(pmy_block);
```

Comment:
Why is the check for FillGhost not required any more?
Would this lead to allocation of comms for all variables?

Reply:
I've moved the check inside `CellVariable::allocateComms`. This function now does the following:
1. Allocate fluxes if and only if `Metadata::WithFluxes` is set
2. Allocate `CellCenteredBoundaryVariable` if and only if `Metadata::FillGhost` is set

Previously, `allocateComms` was only called if `Metadata::FillGhost` was true, so that means if `WithFluxes` was set but `FillGhost` was not, the fluxes would *not* be allocated. Now they are. That's what we want, right?

Reply:
I just pushed another commit to rename `allocateComms` to something more descriptive.